release/release.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/mongodb/mongo-tools/release/platform"
+)
+
+func main() {
+	if len(os.Args) != 2 {
+		fail("please provide exactly one subcommand name")
+	}
+	cmd := os.Args[1]
+
+	switch cmd {
+	case "build-archive":
+		buildTarball()
+		buildZip()
+	case "build-installer":
+		fail("not implemented")
+	case "build-linux":
+		fail("not implemented")
+	}
+}
+
+var binaries = []string{
+	"bsondump",
+	"mongodump",
+	"mongoexport",
+	"mongofiles",
+	"mongoimport",
+	"mongorestore",
+	"mongostat",
+	"mongotop",
+}
+
+var staticFiles = []string{
+	"LICENSE.md",
+	"README.md",
+	"THIRD-PARTY-NOTICES",
+}
+
+func log(format string, args ...interface{}) {
+	fmt.Fprintf(os.Stderr, format+"\n", args...)
+}
+
+func check(err error, format ...interface{}) {
+	if err == nil {
+		return
+	}
+	msg := err.Error()
+	if len(format) != 0 {
+		task := fmt.Sprintf(format[0].(string), format[1:]...)
+		msg = fmt.Sprintf("'%s' failed: %v", task, err)
+	}
+	fail(msg)
+}
+
+func fail(format string, args ...interface{}) {
+	log(format, args...)
+	os.Exit(1)
+}
+
+func run(name string, args ...string) (string, error) {
+	cmd := exec.Command(name, args...)
+	out, err := cmd.Output()
+	return strings.TrimSpace(string(out)), err
+}
+
+func getVersion() string {
+	desc, err := run("git", "describe")
+	check(err, "git describe")
+	return desc
+}
+
+func getReleaseName() string {
+	p, err := platform.Get()
+	check(err, "get platform")
+	version := getVersion()
+
+	return fmt.Sprintf(
+		"mongodb-cli-tools-%s-%s-%s",
+		p.Name, p.Arch, version,
+	)
+}
+
+func addToTarball(tw *tar.Writer, dst, src string) {
+	file, err := os.Open(src)
+	check(err, "open file")
+	defer file.Close()
+
+	stat, err := file.Stat()
+	check(err, "stat file")
+
+	header := &tar.Header{
+		Name: dst,
+		Size: stat.Size(),
+		Mode: int64(stat.Mode()),
```

Comment:
Do we want to force this to something like 0755 rather than rely on what's on the filesystem?

Reply:
done!